homeassistant/components/glances/sensor.py

```diff
@@ -302,10 +295,19 @@ def _handle_coordinator_update(self) -> None:
 
     def _update_native_value(self) -> None:
         """Update sensor native value from coordinator data."""
-        data = self.coordinator.data[self.entity_description.type]
-        if dict_val := data.get(self._sensor_label):
+        data = self.coordinator.data.get(self.entity_description.type)
```

Comment:
```suggestion
        data = self.coordinator.data.get(self.entity_description.type)
        if not data:
            self._attr_available = False
            return
```
Then the rest of changes below can be changed back.
Also not sure why it's needed with all this extensive checking for available in the new method you made.

Reply:
It's not really equivalent as this would only test the absence for a whole module (like fs).
But in glances, some items inside a module can also be removed server side. For example, glances can be configured not to report one of several file systems (like /boot). This would still provoke a missing key exception.
So I think the `else` clause at the end is the safest.

Those checks are needed because they happen during data update, before any availability check. In this PR, we handle the case where Glances answered with some data (no error) but not all data (some modules or keys were removed server side).

For the available method:
I only added a single check on None value to the existing code which was previously in the available() method.
I didn't want to change that logic, but from what I understand Glances can provide a string in a numeric sensor (for example "unknown"). In that case those checks would flag the sensor as unavailable.